### PR TITLE
Add download to memory functionality

### PR DESCRIPTION
**Pull request by sluenenglish**
- for workbooks and datasources
- existing download functionality renamed to `download_to_file`

### CODE REVIEW

**Comment by sluenenglish** [comment]
Needs tests

File: tableauserverclient/server/endpoint/datasources_endpoint.py [hunk 1]
```diff
@@ -66,7 +66,7 @@ def delete(self, datasource_id):
     # Download 1 datasource by id
     @api(version="2.0")
     @parameter_added_in(no_extract='2.5')
-    def download(self, datasource_id, filepath=None, no_extract=False):
```
Comment (VathsalaAchar) [inline tableauserverclient/server/endpoint/datasources_endpoint.py:69]: Any reason why the method name needs to change?

**Comment by VathsalaAchar** [comment]
Could you change the base to be the atheon branch rather than development? Unless you want to create a PR against the tableau development branch?

And let's discuss the change core method names; I have a feeling this might make it harder to merge in to tableau.

File: tableauserverclient/server/endpoint/datasources_endpoint.py (hunk 1 above)
Comment (sluenenglish) [reply tableauserverclient/server/endpoint/datasources_endpoint.py:69]: I have changed this back to the original method name

**Pull request merged**